### ISSUE TITLE
Alpine build from source with gcc errors.

### ISSUE THREAD

**Issue by gcoakes** [ziglang/zig#12407]
### Zig Version

74673b7f69b27dc39a653f92eb58bba71e289f39

### Steps to Reproduce

Launch an Alpine edge container:

``` console
$ podman run -it --rm -v "$PWD:/src:z" alpine:edge
```

Within the alpine container, compile zig from source:

``` console
$ apk add git cmake make gcc g++ clang clang-libs clang-dev llvm llvm-dev lld lld-dev
$ mkdir build
$ cd build
$ cmake '-DCLANG_LIBRARIES=/usr/lib/libclang.so;/usr/lib/libclang-cpp.so' -DCLANG_INCLUDE_DIRS=/usr/lib/clang/14.0.6/include/ ..
$ make
```

### Expected Behavior

Compiles zig from source using gcc and linking to Alpine's packaged llvm libraries.

### Actual Behavior

``` console
$ podman run -it --rm -v "$PWD:/src:z" alpine:edge
/ # apk add git cmake make gcc g++ clang clang-libs clang-dev llvm llvm-dev lld lld-dev
fetch https://dl-cdn.alpinelinux.org/alpine/edge/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/edge/community/x86_64/APKINDEX.tar.gz
(1/58) Installing libffi (3.4.2-r1)
(2/58) Installing libgcc (11.2.1_git20220219-r5)
(3/58) Installing libstdc++ (11.2.1_git20220219-r5)
(4/58) Installing xz-libs (5.2.5-r1)
(5/58) Installing libxml2 (2.9.14-r1)
(6/58) Installing llvm14-libs (14.0.6-r3)
(7/58) Installing clang-libs (14.0.6-r1)
(8/58) Installing clang (14.0.6-r1)
(9/58) Installing clang-extra-tools (14.0.6-r1)
(10/58) Installing clang-dev (14.0.6-r1)
(11/58) Installing libacl (2.3.1-r1)
(12/58) Installing libbz2 (1.0.8-r2)
(13/58) Installing libcrypto3 (3.0.5-r1)
(14/58) Installing libexpat (2.4.8-r1)
(15/58) Installing lz4-libs (1.9.3-r1)
(16/58) Installing zstd-libs (1.5.2-r5)
(17/58) Installing libarchive (3.6.1-r1)
(18/58) Installing ca-certificates (20220614-r2)
(19/58) Installing brotli-libs (1.0.9-r7)
(20/58) Installing nghttp2-libs (1.48.0-r1)
(21/58) Installing libssl3 (3.0.5-r1)
(22/58) Installing libcurl (7.84.0-r2)
(23/58) Installing ncurses-terminfo-base (6.3_p20220806-r0)
(24/58) Installing ncurses-libs (6.3_p20220806-r0)
(25/58) Installing rhash-libs (1.4.3-r1)
(26/58) Installing libuv (1.44.2-r0)
(27/58) Installing cmake (3.24.0-r0)
(28/58) Installing binutils (2.38-r4)
(29/58) Installing libgomp (11.2.1_git20220219-r5)
(30/58) Installing libatomic (11.2.1_git20220219-r5)
(31/58) Installing gmp (6.2.1-r2)
(32/58) Installing isl24 (0.24-r0)
(33/58) Installing mpfr4 (4.1.0-r0)
(34/58) Installing mpc1 (1.2.1-r0)
(35/58) Installing gcc (11.2.1_git20220219-r5)
(36/58) Installing musl-dev (1.2.3-r1)
(37/58) Installing libc-dev (0.7.2-r3)
(38/58) Installing g++ (11.2.1_git20220219-r5)
(39/58) Installing pcre2 (10.40-r0)
(40/58) Installing git (2.37.1-r1)
(41/58) Installing lld-libs (14.0.6-r4)
(42/58) Installing lld (14.0.6-r4)
(43/58) Installing lld-dev (14.0.6-r4)
(44/58) Installing llvm14 (14.0.6-r3)
(45/58) Installing gdbm (1.23-r0)
(46/58) Installing mpdecimal (2.5.1-r1)
(47/58) Installing readline (8.1.2-r0)
(48/58) Installing sqlite-libs (3.39.2-r0)
(49/58) Installing python3 (3.10.5-r2)
(50/58) Installing py3-appdirs (1.4.4-r4)
(51/58) Installing py3-more-itertools (8.13.0-r0)
(52/58) Installing py3-ordered-set (4.0.2-r4)
(53/58) Installing py3-parsing (3.0.9-r0)
(54/58) Installing py3-packaging (21.3-r2)
(55/58) Installing py3-setuptools (59.4.0-r0)
(56/58) Installing llvm14-test-utils (14.0.6-r3)
(57/58) Installing llvm14-dev (14.0.6-r3)
(58/58) Installing make (4.3-r0)
Executing busybox-1.35.0-r18.trigger
Executing ca-certificates-20220614-r2.trigger
OK: 699 MiB in 72 packages
/ # cd /src
/src # git describe --always
74673b7f6
/src # git clean -xfd
/src # git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
/src # mkdir build
/src # cd build/
/src/build # cmake '-DCLANG_LIBRARIES=/usr/lib/libclang.so;/usr/lib/libclang-cpp.so' -DCLANG_INCLUDE_DIRS=/usr/lib/clang/14.0.6/include/ ..
-- The C compiler identification is GNU 11.2.1
-- The CXX compiler identification is GNU 11.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring zig version 0.10.0
-- Found llvm: -lLLVM-14 (Required is at least version "14")
-- Found clang: /usr/lib/libclang.so;/usr/lib/libclang-cpp.so (Required is at least version "14")
-- Found lld: /usr/lib/liblldMinGW.so;/usr/lib/liblldELF.so;/usr/lib/liblldCOFF.so;/usr/lib/liblldWasm.so;/usr/lib/liblldMachO.so;/usr/lib/liblldCommon.so (Required is at least version "14")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Configuring done
-- Generating done
-- Build files have been written to: /src/build
/src/build # make
[  0%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/f128M_isSignalingNaN.c.o
[  1%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/extF80M_isSignalingNaN.c.o
[  1%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/s_commonNaNToF128M.c.o
[  2%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/s_commonNaNToExtF80M.c.o
[  2%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/s_commonNaNToF16UI.c.o
[  3%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/s_commonNaNToF32UI.c.o
[  3%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/s_commonNaNToF64UI.c.o
[  4%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/s_f128MToCommonNaN.c.o
[  4%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/s_extF80MToCommonNaN.c.o
[  5%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/s_f16UIToCommonNaN.c.o
[  5%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/s_f32UIToCommonNaN.c.o
[  6%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/s_f64UIToCommonNaN.c.o
[  6%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/s_propagateNaNF128M.c.o
[  7%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/s_propagateNaNExtF80M.c.o
[  7%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/s_propagateNaNF16UI.c.o
[  8%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/8086/softfloat_raiseFlags.c.o
[  8%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_add.c.o
[  9%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_div.c.o
[  9%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_eq.c.o
[ 10%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_eq_signaling.c.o
[ 10%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_le.c.o
[ 11%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_le_quiet.c.o
[ 11%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_lt.c.o
[ 12%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_lt_quiet.c.o
[ 12%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_mul.c.o
[ 13%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_mulAdd.c.o
[ 13%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_rem.c.o
[ 14%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_roundToInt.c.o
[ 14%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_sqrt.c.o
[ 15%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_sub.c.o
[ 15%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_to_f16.c.o
[ 16%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_to_f32.c.o
[ 17%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_to_f64.c.o
[ 17%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_to_extF80M.c.o
[ 18%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_to_i32.c.o
[ 18%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_to_i32_r_minMag.c.o
[ 19%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_to_i64.c.o
[ 19%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_to_i64_r_minMag.c.o
[ 20%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_to_ui32.c.o
[ 20%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_to_ui32_r_minMag.c.o
[ 21%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_to_ui64.c.o
[ 21%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f128M_to_ui64_r_minMag.c.o
[ 22%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_add.c.o
[ 22%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_div.c.o
[ 23%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_eq.c.o
[ 23%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_le.c.o
[ 24%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_lt.c.o
[ 24%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_mul.c.o
[ 25%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_rem.c.o
[ 25%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_roundToInt.c.o
[ 26%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_sqrt.c.o
[ 26%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_sub.c.o
[ 27%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_to_f16.c.o
[ 27%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_to_f32.c.o
[ 28%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_to_f64.c.o
[ 28%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/extF80M_to_f128M.c.o
[ 29%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_add.c.o
[ 29%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_div.c.o
[ 30%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_eq.c.o
[ 30%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_isSignalingNaN.c.o
[ 31%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_lt.c.o
[ 31%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_mul.c.o
[ 32%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_rem.c.o
[ 32%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_roundToInt.c.o
[ 33%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_sqrt.c.o
[ 34%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_sub.c.o
[ 34%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_to_extF80M.c.o
[ 35%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_to_f128M.c.o
[ 35%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_to_f64.c.o
[ 36%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f32_to_extF80M.c.o
[ 36%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f32_to_f128M.c.o
[ 37%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f64_to_extF80M.c.o
[ 37%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f64_to_f128M.c.o
[ 38%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f64_to_f16.c.o
[ 38%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/i32_to_f128M.c.o
[ 39%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_add256M.c.o
[ 39%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_addCarryM.c.o
[ 40%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_addComplCarryM.c.o
[ 40%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_addF128M.c.o
[ 41%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_addExtF80M.c.o
[ 41%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_addM.c.o
[ 42%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_addMagsF16.c.o
[ 42%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_addMagsF32.c.o
[ 43%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_addMagsF64.c.o
[ 43%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_approxRecip32_1.c.o
[ 44%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_approxRecipSqrt32_1.c.o
[ 44%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_approxRecipSqrt_1Ks.c.o
[ 45%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_approxRecip_1Ks.c.o
[ 45%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_compare128M.c.o
[ 46%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_compare96M.c.o
[ 46%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_compareNonnormExtF80M.c.o
[ 47%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_countLeadingZeros16.c.o
[ 47%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_countLeadingZeros32.c.o
[ 48%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_countLeadingZeros64.c.o
[ 48%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_countLeadingZeros8.c.o
[ 49%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_eq128.c.o
[ 50%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_invalidF128M.c.o
[ 50%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_invalidExtF80M.c.o
[ 51%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_isNaNF128M.c.o
[ 51%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_le128.c.o
[ 52%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_lt128.c.o
[ 52%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_mul128MTo256M.c.o
[ 53%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_mul64To128M.c.o
[ 53%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_mulAddF128M.c.o
[ 54%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_mulAddF16.c.o
[ 54%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_mulAddF32.c.o
[ 55%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_mulAddF64.c.o
[ 55%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_negXM.c.o
[ 56%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_normExtF80SigM.c.o
[ 56%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_normRoundPackMToF128M.c.o
[ 57%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_normRoundPackMToExtF80M.c.o
[ 57%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_normRoundPackToF16.c.o
[ 58%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_normRoundPackToF32.c.o
[ 58%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_normRoundPackToF64.c.o
[ 59%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_normSubnormalF128SigM.c.o
[ 59%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_normSubnormalF16Sig.c.o
[ 60%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_normSubnormalF32Sig.c.o
[ 60%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_normSubnormalF64Sig.c.o
[ 61%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_remStepMBy32.c.o
[ 61%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_roundMToI64.c.o
[ 62%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_roundMToUI64.c.o
[ 62%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_roundPackMToExtF80M.c.o
[ 63%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_roundPackMToF128M.c.o
[ 63%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_roundPackToF16.c.o
[ 64%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_roundPackToF32.c.o
[ 64%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_roundPackToF64.c.o
[ 65%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_roundToI32.c.o
[ 65%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_roundToI64.c.o
[ 66%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_roundToUI32.c.o
[ 67%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_roundToUI64.c.o
[ 67%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_shiftLeftM.c.o
[ 68%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_shiftNormSigF128M.c.o
[ 68%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_shiftRightJam256M.c.o
[ 69%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_shiftRightJam32.c.o
[ 69%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_shiftRightJam64.c.o
[ 70%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_shiftRightJamM.c.o
[ 70%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_shiftRightM.c.o
[ 71%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_shortShiftLeft64To96M.c.o
[ 71%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_shortShiftLeftM.c.o
[ 72%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_shortShiftRightExtendM.c.o
[ 72%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_shortShiftRightJam64.c.o
[ 73%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_shortShiftRightJamM.c.o
[ 73%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_shortShiftRightM.c.o
[ 74%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_sub1XM.c.o
[ 74%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_sub256M.c.o
[ 75%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_subM.c.o
[ 75%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_subMagsF16.c.o
[ 76%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_subMagsF32.c.o
[ 76%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_subMagsF64.c.o
[ 77%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_tryPropagateNaNF128M.c.o
[ 77%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/s_tryPropagateNaNExtF80M.c.o
[ 78%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/f16_mulAdd.c.o
[ 78%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/softfloat_state.c.o
[ 79%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/ui32_to_f128M.c.o
[ 79%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/ui64_to_f128M.c.o
[ 80%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/ui32_to_extF80M.c.o
[ 80%] Building C object CMakeFiles/embedded_softfloat.dir/deps/SoftFloat-3e/source/ui64_to_extF80M.c.o
[ 81%] Linking C static library zigcpp/libembedded_softfloat.a
[ 81%] Built target embedded_softfloat
[ 81%] Building CXX object CMakeFiles/zigcpp.dir/src/zig_llvm.cpp.o
In file included from /src/src/zig_llvm.h:12,
                 from /src/src/zig_llvm.cpp:16:
/usr/lib/clang/14.0.6/include/stddef.h:18:19: error: missing binary operator before token "("
   18 | #if !__has_feature(modules)
      |                   ^
/usr/lib/clang/14.0.6/include/stddef.h:30:42: error: missing binary operator before token "("
   30 | #if !defined(_PTRDIFF_T) || __has_feature(modules)
      |                                          ^
/usr/lib/clang/14.0.6/include/stddef.h:41:39: error: missing binary operator before token "("
   41 | #if !defined(_SIZE_T) || __has_feature(modules)
      |                                       ^
/usr/lib/clang/14.0.6/include/stddef.h:55:42: error: missing binary operator before token "("
   55 |      !defined(_RSIZE_T)) || __has_feature(modules)
      |                                          ^
make[2]: *** [CMakeFiles/zigcpp.dir/build.make:76: CMakeFiles/zigcpp.dir/src/zig_llvm.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:119: CMakeFiles/zigcpp.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Comment by andrewrk** [comment]
This is not a problem with zig; your cmake line is incorrectly configured. Please follow the official directions for building from source

**Comment by gcoakes** [comment]
I already did my due diligence in reviewing the [Build Zig From Source](https://github.com/ziglang/zig/wiki/Building-Zig-From-Source) document, [Troubleshooting Build Issues](https://github.com/ziglang/zig/wiki/Troubleshooting-Build-Issues), and consulting with members of Discord/element.io. That cmake command was the closest I got to compiling properly. The first thing I attempted was `cmake ..` which gave errors and possible solutions:

```
...
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
CLANG_INCLUDE_DIRS (ADVANCED)
   used as include directory in directory /src
   used as include directory in directory /src
   used as include directory in directory /src
   used as include directory in directory /src
   used as include directory in directory /src
   used as include directory in directory /src
   used as include directory in directory /src
   used as include directory in directory /src
   used as include directory in directory /src
   used as include directory in directory /src
   used as include directory in directory /src
   used as include directory in directory /src
CLANG_LIBRARIES (ADVANCED)
    linked by target "zigcpp" in directory /src
```

That's how I ended up with the cmake arguments from the original post, and I suspect that is also how #2608 happened as well.

I eventually took the long route and compiled llvm from source and used clang as the CC and CXX; however, I filed this issue in the hopes there might be some productive, documented discussion specifically involving the `__has_feature` error message. The only other occurrence of that error message was similarly dismissed without resolution.

Having a fresh set of eyes on the issue tonight, I discovered the expectation is for you to ignore the error output of `cmake ..` and instead run `cmake -DCMAKE_PREFIX_PATH=/usr ..`. This is documented in the Build Zig From Source but obviously not sufficiently for me to parse out on the first few reads.

@andrewrk, I really wish this issue and #2608 were not so quickly dismissed. I'm not certain what kind of business pressure might be coming down on you to close out issues so quickly without resolution, but I think there might be a better way to handle it. Perhaps you might consider leaving issues like this open for a few days so that other contributors have a chance to respond?
